tests/models/vivit/test_modeling_vivit.py

```diff
@@ -0,0 +1,350 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Testing suite for the PyTorch ViViT model. """
+
+
+import copy
+import inspect
+import unittest
+
+import numpy as np
+from huggingface_hub import hf_hub_download
+
+from transformers import VivitConfig
+from transformers.models.auto import get_values
+from transformers.testing_utils import require_torch, require_vision, slow, torch_device
+from transformers.utils import cached_property, is_torch_available, is_vision_available
+
+from ...test_configuration_common import ConfigTester
+from ...test_modeling_common import ModelTesterMixin, floats_tensor, ids_tensor
+
+
+if is_torch_available():
+    import torch
+    from torch import nn
+
+    from transformers import MODEL_FOR_VIDEO_CLASSIFICATION_MAPPING, VivitForVideoClassification, VivitModel
+    from transformers.models.vivit.modeling_vivit import VIVIT_PRETRAINED_MODEL_ARCHIVE_LIST
+
+
+if is_vision_available():
+    from transformers import VivitImageProcessor
+
+
+class VivitModelTester:
+    def __init__(
+        self,
+        parent,
+        batch_size=2,
+        is_training=True,
+        use_labels=True,
+        num_labels=10,
+        image_size=224,
+        num_frames=8,  # decreased, because default 32 takes too much RAM at inference
+        tubelet_size=[2, 16, 16],
+        num_channels=3,
+        hidden_size=768,
+        num_hidden_layers=12,
+        num_attention_heads=12,
+        intermediate_size=3072,
```

Comment:
One thing to note here is that many of these values are very large and will cause the tests to run slowly. Could you update `image_size`, `num_frames`, `tubelete_size`, `hidden_size`, `num_attention_heads`, `num_hidden_layers` and `intermediate_size` to all be as small as possible? You can look at other model's test files to find typical values to use.

Reply:
thanks! i've decreased those values and the tests pass now!